examples/dataservice-example/main.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <getopt.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <iostream>
+
+#include "Examples.h"
+
+const AccessKey access_key{
+    "", ""};  // put your here.access.key.id and here.access.key.secret
+
+constexpr auto usage =
+    "usuage is \n -a [--all] : run all examples \n -e [--example] : run "
+    "example [read, write, cache] \n -h [--help]: show usage ";
+
+int RunAllExamples() {
+  std::cout << "Run all examples" << std::endl;
+  std::cout << "Read Example" << std::endl;
+  if (RunExampleRead(access_key)) {
+    std::cout << "Read Example failed" << std::endl;
+    return -1;
+  }
+
+  std::cout << "Write example" << std::endl;
+  if (RunExampleWrite(access_key)) {
+    std::cout << "Write Example failed" << std::endl;
+    return -1;
+  }
+  std::cout << "Protected cache example" << std::endl;
+  if (RunExampleProtectedCache()) {
+    std::cout << "Protected cache Example failed" << std::endl;
+    return -1;
+  }
+  return 0;
+}
+
+int main(int argc, char** argv) {
+  static struct option long_options[] = {{"example", required_argument, 0, 'e'},
+                                         {"all", no_argument, 0, 'a'},
+                                         {"help", no_argument, 0, 'h'},
+                                         {0, 0, 0, 0}};
+
+  int opt, option_index;
+  while ((opt = getopt_long(argc, argv, "e:ah", long_options, &option_index)) !=
+         EOF)
+    switch (opt) {
+      case 'e': {
```

Comment:
```suggestion
      case 'e':
```
Sorry for the caused confusion. What I meant in the previous review is to use curly-braces for the `if` statements not for this `case` block.

Reply:
updated, can you take a look